client/src/containers/Dashboard.jsx

```diff
@@ -432,8 +433,7 @@ class Dashboard extends Component {
 
   handleToggleShareProtected(isProtected) {
     this.setState({ passwordAlert: null });
-    const dashboard = getDashboardFromState(this.state.dashboard, true);
-    this.props.dispatch(actions.setShareProtection(dashboard.shareId, { protected: isProtected }));
```

Comment:
also why are we removing this API request? Isnt this how we save the config?

Reply:
This get's called when we enable the radio button & yet have to type a password. The backend will respond with an error message (no password) and the client will see the error. This removes the initial request. And the user has to hit "save" once he/she entered a valid password. At that point a request with all config necessary to enable protection is available.